sklearn/feature_extraction/_hashing.pyx

```diff
@@ -103,4 +105,17 @@ def transform(raw_X, Py_ssize_t n_features, dtype, bint alternate_sign=1):
     else:
         indptr_a = indptr_a.astype(np.int32)
 
-    return (indices_a, indptr_a, values[:size])
+    return (indices_a, indptr_a, values[:size], feature_to_index_map)
+
+
+def _save_feature_mapping(into_dict, key, value):
```

Comment:
We can't afford this expense unless the user explicitly says they want this stored. we need a constructor parameter to control that.

Reply:
Got it

Reply:
should I raise a warning message as well?

Reply:
What for?

Reply:
That using save_mappings=True may lower the performance? or is it not required?

Reply:
I've added it in docstring though